doc/aaa/ldap.md

```diff
@@ -0,0 +1,2617 @@
+     |           |  +--rw oc-aaa-ldap-ext:retransmit-attempts?          uint8
+     |           |  +--rw oc-aaa-ldap-ext:port?                         oc-inet:port-number
+     |           |  +--rw oc-aaa-ldap-ext:version?                      uint8
+     |           |  +--rw oc-aaa-ldap-ext:base?                         string
+     |           |  +--rw oc-aaa-ldap-ext:ssl?                          ldap-ssl-type
+     |           |  +--rw oc-aaa-ldap-ext:bind-dn?                      string
+     |           |  +--rw oc-aaa-ldap-ext:bind-pw?                      string
+     |           |  +--rw oc-aaa-ldap-ext:source-address?               oc-inet:ip-address
+     |           |  +--rw oc-aaa-ldap-ext:vrf-name?                     string
+     |           |  +--rw oc-aaa-ldap-ext:idle-time-limit?              uint32
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-group?               string
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-shadow?              string
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-netgroup?            string
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-sudoers?             string
+     |           |  +--rw oc-aaa-ldap-ext:nss-initgroups-ignoreusers?   string
+     |           |  +--rw oc-aaa-ldap-ext:pam-filter?                   string
+     |           |  +--rw oc-aaa-ldap-ext:pam-login-attribute?          string
+     |           |  +--rw oc-aaa-ldap-ext:pam-group-dn?                 string
+     |           |  +--rw oc-aaa-ldap-ext:pam-member-attribute?         string
+     |           |  +--rw oc-aaa-ldap-ext:sudoers-base?                 string
+     |           |  +--rw oc-aaa-ldap-ext:scope?                        enumeration
+     |           |  +--rw oc-aaa-ldap-ext:nss-base-passwd?              string
+     |           +--ro oc-aaa-ldap-ext:state
+     |           +--rw oc-aaa-ldap-ext:maps
+     |              +--rw oc-aaa-ldap-ext:map* [name from]
+     |                 +--rw oc-aaa-ldap-ext:name      -> ../config/name
+     |                 +--rw oc-aaa-ldap-ext:from      -> ../config/from
+     |                 +--rw oc-aaa-ldap-ext:config
+     |                 |  +--rw oc-aaa-ldap-ext:name?   enumeration
+     |                 |  +--rw oc-aaa-ldap-ext:from?   string
+     |                 |  +--rw oc-aaa-ldap-ext:to?     string
+     |                 +--ro oc-aaa-ldap-ext:state
+     |                    +--ro oc-aaa-ldap-ext:name?   enumeration
+     |                    +--ro oc-aaa-ldap-ext:from?   string
+     |                    +--ro oc-aaa-ldap-ext:to?     string
+...
+...
+
+
+```
+
+## REST/gNMI
+
+REST, and gNMI support would be needed by writing transformer annotation,
+overloads, and SONiC Yang.
+
+## KLISH
+
+KLISH commands (XML, actioners, renderers) needs to be written for
+
+- AAA authentication ldap option.
+
+```
+sonic(config)# [no] aaa authentication login-method { ldap | local }*
```

Comment:
The way this is written may lead the reader to think that the login method is either ldap OR local. In fact, more than one method may be specified. To ensure consistency with other vendors (primarily Cisco) may I suggest we document the command as follows (this is pretty much how Cisco documents this command and how customers expect the command to work):
```
aaa authentication login CONNECTION METHOD_1 [METHOD_2] ... [METHOD_N]
no aaa authentication login CONNECTION
```
Where:
**CONNECTION** : Connection type of sessions for which authentication list is used
**default**: authentication list for non-console logins (e.g. ssh)
**console**: authentication list for console logins

NOTE: I understand that SONiC currently does not distinguish between console and non-console logins. It's fine to omit the CONNECTION parameter for now, but it's just a mater of time before this is requested.

**METHOD_X** : An authentication​ service.
**group radius**: a server group that consists of all defined RADIUS hosts.
**group tacacs+**: a server group that consists of all defined TACACS+ hosts.
**group ldap**: a server group that consists of all defined LDAP hosts.
**local**: local authentication (i.e. unix)
**none**: no authentication